mountpoint-s3/src/inode.rs

```diff
@@ -68,20 +72,24 @@ struct SuperblockInner {
     inodes: RwLock<HashMap<InodeNo, Inode>>,
     next_ino: AtomicU64,
     mount_time: OffsetDateTime,
+    cache_config: CacheConfig,
 }
 
 impl Superblock {
     /// Create a new Superblock that targets the given bucket/prefix
-    pub fn new(bucket: &str, prefix: &Prefix) -> Self {
+    pub fn new(bucket: &str, prefix: &Prefix, cache_config: CacheConfig) -> Self {
         let mount_time = OffsetDateTime::now_utc();
+
         let root = InodeInner {
             ino: ROOT_INODE_NO,
             parent: ROOT_INODE_NO,
             name: String::new(),
             full_key: prefix.to_string(),
             kind: InodeKind::Directory,
             sync: RwLock::new(InodeState {
-                stat: InodeStat::for_directory(mount_time, Instant::now()), // TODO expiry
+                // The root inode never expires because there's no remote to consult for its
+                // metadata, and it always exists.
+                stat: InodeStat::for_directory(mount_time, NEVER_EXPIRE_TTL),
```

Comment:
The breakdown looks something like this:
* Remote files -- S3 has the authoritative metadata, so we use a short TTL
* Remote directories -- there's not really any metadata we care about, but we do want to check whether the directory still exists, so we use a slightly longer TTL
* Local files -- the metadata is all local, so it should never expire (although open question what we should do if someone creates the remote file while the local file is being written)
* Local directories -- there's not really any metadata we care about, and we don't care whether the directory exists remotely (it doesn't change anything for us), so it never expires

This particular case is for the root directory, which is special because it will never be remote (it's the root of the bucket).